code/modules/vehicles/scooter.dm

```diff
@@ -146,3 +146,44 @@
 		qdel(src)
 	return TRUE
 
+//Wheelys
+/obj/vehicle/ridden/scooter/wheelys
+	name = "Wheely-Heels"
+	desc = "Uses patented retractable wheel technology. Never sacrifice speed for style - not that this provides much of either."
+	icon = null
+	density = FALSE
+
+/obj/vehicle/ridden/scooter/wheelys/Initialize()
+	. = ..()
+	var/datum/component/riding/D = LoadComponent(/datum/component/riding)
+	D.vehicle_move_delay = 0
+	D.set_vehicle_dir_layer(SOUTH, ABOVE_MOB_LAYER)
+	D.set_vehicle_dir_layer(NORTH, OBJ_LAYER)
+	D.set_vehicle_dir_layer(EAST, OBJ_LAYER)
+	D.set_vehicle_dir_layer(WEST, OBJ_LAYER)
+
+/obj/vehicle/ridden/scooter/wheelys/post_unbuckle_mob(mob/living/M)
+	if(!has_buckled_mobs())
+		to_chat(M, "<span class='notice'>You pop the Wheely-Heel's wheels back into place.</span>")
+		loc = null
+	return ..()
+
+/obj/vehicle/ridden/scooter/wheelys/post_buckle_mob(mob/living/M)
+	to_chat(M, "<span class='notice'>You pop out the Wheely-Heel's wheels.</span>")
+	return ..()
+
+/obj/vehicle/ridden/scooter/wheelys/Collide(atom/A)
```

Comment:
not sure why these are a child of scooters instead of skateboards if you're just copying all of the skateboard code here anyways

Reply:
While it has a lot in common with skateboards, its different enough that I'd be overriding so much it wouldn't make sense to make it a child.